### PR TITLE
[SPARK-29080][CORE][SPARKR] Support R file extension case-insensitively

### PR DESCRIPTION
**Pull request by Loquats**
### What changes were proposed in this pull request?

Make r file extension check case insensitive for spark-submit.

### Why are the changes needed?

spark-submit does not accept `.r` files as R scripts. Some codebases have r files that end with lowercase file extensions. It is inconvenient to use spark-submit with lowercase extension R files. The error is not very clear (https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala#L232). 

```
$ ./bin/spark-submit examples/src/main/r/dataframe.r
Exception in thread "main" org.apache.spark.SparkException: Cannot load main class from JAR file:/Users/dongjoon/APACHE/spark-release/spark-2.4.4-bin-hadoop2.7/examples/src/main/r/dataframe.r
```

### Does this PR introduce any user-facing change?

Yes. spark-submit can now be used to run R scripts with `.r` file extension.

### How was this patch tested?

Manual.

```
$ mv examples/src/main/r/dataframe.R examples/src/main/r/dataframe.r
$ ./bin/spark-submit examples/src/main/r/dataframe.r
```

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
To check the coding style quickly, you can use `dev/scalastyle` in your PR directory. Please try to run it and fix the errors. Thanks.

**Comment by Loquats** [comment]
@dongjoon-hyun Thanks for the tip. I ran `dev/scalastyle` and fixed the scalastyle errors. Also tried testing this in `build/sbt` with:
```
project core
test
project launcher
test
```
The tests for these projects pass.
I'm not sure what project the `resource-manager` files are part of, but I ran the following and they also pass:
```
testOnly *ClientSuite
testOnly *ApplicationMasterSuite
```

Please let me know if there is anything else I should test.

**Comment by dongjoon-hyun** [comment]
Thank you for update.

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
@Loquats . BTW, do you have Apache JIRA account? You can create an Apache Spark JIRA issue there. Please file a JIRA issue and use the JIRA ID in the PR title next time.

For this issue, I created one for you and updated this PR title.
- https://issues.apache.org/jira/browse/SPARK-29080

**Comment by Loquats** [comment]
@dongjoon-hyun Thanks for creating the JIRA issue. I just created an Apache JIRA account, and I will remember to create a JIRA issue next time. My username is andyzhang.

**Comment by dongjoon-hyun** [comment]
Thank you for informing your JIRA id. When this PR is merged, your ID will be added to Apache Spark contributor group and SPARK-29080 will be assigned to you.

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala [hunk 1]
```diff
@@ -703,7 +703,8 @@ private[spark] class ApplicationMaster(
       // of files to add to PYTHONPATH, which Client.scala already handles, so it's empty.
       userArgs = Seq(args.primaryPyFile, "") ++ userArgs
     }
-    if (args.primaryRFile != null && args.primaryRFile.endsWith(".R")) {
+    if (args.primaryRFile != null &&
+      (args.primaryRFile.endsWith(".R") || args.primaryRFile.endsWith(".r"))) {
```
Comment (dongjoon-hyun) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala]: We need two more spaces here.

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 2]
```diff
@@ -973,7 +973,8 @@ private[spark] class Client(
       } else {
         Utils.classForName("org.apache.spark.deploy.yarn.ExecutorLauncher").getName
       }
-    if (args.primaryRFile != null && args.primaryRFile.endsWith(".R")) {
+    if (args.primaryRFile != null &&
+      (args.primaryRFile.endsWith(".R") || args.primaryRFile.endsWith(".r"))) {
```
Comment (dongjoon-hyun) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: We need two more space here for indentation.

**Review by felixcheung — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
I updated the PR description for the others because this will become a commit log.
Thank you for your first contribution, @Loquats . 
Merged to master.

**Comment by dongjoon-hyun** [comment]
You are added to the Apache Spark contributor group and SPARK-29080 is assigned to you.

**Review by HyukjinKwon — COMMENTED** [review]
Lgtm too

**Comment by Loquats** [comment]
Thanks all for reviewing this PR!

**Pull request closed without merging**